io.openems.edge.evcs.goe.chargerhome/src/io/openems/edge/evcs/goe/chargerhome/GoeAPI.java

```diff
@@ -0,0 +1,259 @@
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.common.utils.JsonUtils;
+
+public class GoeAPI {
+	
+	private final String IP;
+	private boolean debugMode;
+	private String debugFile;
+	private int executeEveryCycle;
+	private int cycle;
+	private JsonObject jsonStatus;
+	private GoeChargerHomeImpl parent;
+
+	public GoeAPI(String ip, boolean debugMode, String debugFile, int StatusAfterCycles, GoeChargerHomeImpl p) {
+		this.IP = ip;
+		this.debugMode = debugMode;
+		this.debugFile = debugFile;
+		this.cycle = 0;
+		this.executeEveryCycle = StatusAfterCycles;
+		this.jsonStatus = null;
+		this.parent = p;
+	}
+
+	/**
+	 * Gets the status from go-e
+	 * 
+	 * See https://github.com/goecharger
+	 * 
+	 * @return the boolean value
+	 * @throws OpenemsNamedException on error
+	 */
+	public JsonObject getStatus() {
+			
+		try {
+			// Execute every x-Cycle
+			if (this.cycle == 0 || this.cycle % this.executeEveryCycle == 0) {
+				JsonObject json = new JsonObject();
+				if (!debugMode) {
+					String URL = "http://" + this.IP + "/status";
+					json = this.sendRequest(URL, "GET");
+				}						
+			
+				if (debugMode) {
+					JsonParser parser = new JsonParser();
+					Object obj = parser.parse(new FileReader(debugFile));
+					json = (JsonObject) obj;
+				}										
+				
+				this.cycle = 1;
+				this.jsonStatus = json;
+				return json;		
+			}
+			else {
+				this.cycle++;
+				return this.jsonStatus;
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	/** Sets the activation status for go-e
+	 * 
+	 * See https://github.com/goecharger
+	 * 
+	 * @return JsonObject with new settings
+	 * @throws OpenemsNamedException on error
+	 */
+	public JsonObject setActive(boolean active) {
+			
+		try {
+			if (active != this.parent.Active) {
+				JsonObject json = new JsonObject();
+				if (!debugMode) {
+					Integer status = 0;
+					if (active) {
+						status = 1;
+					}
+					String URL = "http://" + this.IP + "/mqtt?payload=alw=" + Integer.toString(status);
+					json = this.sendRequest(URL, "PUT");
+					this.parent.Active = active;					
+					return json;
+				}		
+				else {
+					this.parent.Active = active;
+					return this.getStatus();
+				}
+			}
+			else {
+				return this.jsonStatus;
+			}			
+
+		} catch (Exception e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	/** Sets the Current in Ampere for go-e
+	 * 
+	 * See https://github.com/goecharger
+	 * 
+	 * @return JsonObject with new settings
+	 * @throws OpenemsNamedException on error
+	 */
+	public JsonObject setCurrent(int current) {
+			
+		try {
+			if (current != this.parent.activeCurrent) {
+				JsonObject json = new JsonObject();
+				if (!debugMode) {
+					Integer CurrentAmpere = current / 1000;
+					String URL = "http://" + this.IP + "/mqtt?payload=amp=" + Integer.toString(CurrentAmpere);
+					json = this.sendRequest(URL, "PUT");
+					this.parent.activeCurrent = current;
+					return json;
```

Comment:
Please make sure that the results are set if there is no communication error happening. 

this.parent._setSetChargePowerLimit(power);
(This is setting the normal value of that channel and not the write value, to store the last power setpoint)

This would be an interesting part, what we could really log, if the debug mode is activated
this.parent.debugLog(result.toString());

This could be also done after you call the method, when the json you return includes a success message or you return a boolean.